pkg/converter/converter.go

```diff
@@ -104,21 +113,30 @@ func (cvt *Converter) Convert(ctx context.Context, source, target string) error
 			logger.Infof("try to pull with plain HTTP for %s", source)
 			cvt.provider.UsePlainHTTP()
 			if err := cvt.pull(ctx, source); err != nil {
-				return errors.Wrap(err, "try to pull image")
+				return metric, errors.Wrap(err, "try to pull image")
 			}
 		} else {
-			return errors.Wrap(err, "pull image")
+			return metric, errors.Wrap(err, "pull image")
 		}
 	}
-	logger.Infof("pulled image %s, elapse %s", source, time.Since(start))
+	metric.SourcePullElapsed = time.Since(start)
+	logger.Infof("pulled image %s, elapse %s", source, metric.SourcePullElapsed)
 
 	logger.Infof("converting image %s", source)
 	start = time.Now()
 	desc, err := cvt.driver.Convert(ctx, cvt.provider, source)
 	if err != nil {
-		return errors.Wrap(err, "convert image")
+		return metric, errors.Wrap(err, "convert image")
+	}
+	metric.ConversionElapsed = time.Since(start)
+	logger.Infof("converted image %s, elapse %s", target, metric.ConversionElapsed)
+	descs, err := images.ChildrenHandler(cvt.provider.ContentStore())(ctx, *desc)
```

Comment:
Can we move the size calculation to `metric.ImageSize(ctx context.Context, cs content.Store, desc ocispec.Descriptor)`.

Reply:
fix